sketch_main.cpp

```diff
@@ -59,174 +27,75 @@ static bool ValidateInput(const char * /*unused*/, const std::string &value) {
 }
 DEFINE_validator(i, &ValidateInput);
 
-static bool ValidateOutput(const char * /*unused*/, const std::string &value) {
-    if (value.empty()) {
-        FLAGS_o = FLAGS_i + "." + FLAGS_sketch_method;
-    }
-    return true;
-}
-DEFINE_validator(o, &ValidateOutput);
-
-void adjust_short_names() {
-    if (!gflags::GetCommandLineFlagInfoOrDie("m").is_default) {
-        FLAGS_sketch_method = FLAGS_m;
-    }
-    if (!gflags::GetCommandLineFlagInfoOrDie("k").is_default) {
-        FLAGS_kmer_length = FLAGS_k;
-    }
-    if (!gflags::GetCommandLineFlagInfoOrDie("f").is_default) {
-        FLAGS_input_format = FLAGS_f;
-    }
-    if (!gflags::GetCommandLineFlagInfoOrDie("w").is_default) {
-        FLAGS_window_size = FLAGS_w;
-    }
-    if (!gflags::GetCommandLineFlagInfoOrDie("s").is_default) {
-        FLAGS_stride = FLAGS_s;
-    }
-}
 
 using namespace ts;
 
-template <typename seq_type, class kmer_type, class embed_type>
-class SketchHelper {
-  public:
-    SketchHelper(
-            std::function<std::vector<embed_type>(const std::vector<kmer_type> &)> sketcher,
-            std::function<Vec2D<double>(const std::vector<uint64_t> &)> slide_sketcher)
-        : sketcher(std::move(sketcher)), slide_sketcher(std::move(slide_sketcher)) {}
-
-    void compute_sketches() {
-        size_t num_seqs = seqs.size();
-        sketches = new3D<embed_type>(seqs.size(), FLAGS_embed_dim, 0);
-        for (size_t si = 0; si < num_seqs; si++) {
-            std::vector<kmer_type> kmers
-                    = seq2kmer<seq_type, kmer_type>(seqs[si], FLAGS_kmer_length, alphabet_size);
-
-            for (size_t i = 0; i < kmers.size(); i += FLAGS_stride) {
-                auto end = std::min(kmers.begin() + i + FLAGS_window_size, kmers.end());
-                std::vector<kmer_type> kmer_slice(kmers.begin() + i, end);
-                std::vector<embed_type> embed_slice = sketcher(kmer_slice);
-                for (int m = 0; m < FLAGS_embed_dim; m++) {
-                    sketches[si][m].push_back(embed_slice[m]);
-                }
+namespace fs = std::filesystem;
+using Tensor23 = std::pair<std::vector<uint32_t>, std::vector<uint32_t>>;
+Tensor23 tensor23(std::vector<uint8_t> &seq) {
+    Tensor23 result({ std::vector<uint32_t>(16), std::vector<uint32_t>(64) });
+    std::transform(seq.begin(), seq.end(), seq.begin(), [](uint8_t a) { return (a > 3) ? 0 : a; });
```

Comment:
what's this line doing?

Reply:
Eliminating potential garbage characters in the DNA string. There probably are none, but this is to be on the safe side.